### ISSUE TITLE
[Tag] Tags with interactions should have button roles and native button interactions

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2868]
### Provide a general summary of the issue here

_Overflow_ and _Removable_ variants have interactions, but they don't have proper roles. Every interactable element should have a programmatic role. They do have a description "Press the backspace or delete the items", but this does not include a role, though it does make it clearer for some users that the item is interactable. 

The only way to remove the items is through using backspace or delete, or tapping/clicking on the "X" button, which is aria-hidden. Thus some screen reader users without a keyboard (e.g., on mobile) cannot remove the items.

### 🤔 Expected Behavior?

Screen readers announce the tag as "button", indicating an interactable element, along with its visible label.

Users can delete the item by activating it e.g., double tapping).

### 😯 Current Behavior

Users hear "Label, press the delete or [...]" but no role. Some screen readers might ignore the description because of the lack of role.

Users on mobile cannot perform the delete action without using a keyboard.

### 💁 Possible Solution

1. Add role button to the tags.
2. Add native button interactions (mouse click, enter and space keys) to cover all styles of use.

Note: Having the backspace or delete keys as options is okay.

### 🖥️ Steps to Reproduce

1. Focus the item with a screen reader -> reads label, and (sometimes) the description
2. (Mobile) perform action (e.g., double tap) -> item does not get removed.

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning

**Comment by Diaan** [comment]
Related to: #2861
